domain/src/main/kotlin/no/nav/su/se/bakover/domain/vilkår/Vilkår.kt

```diff
@@ -28,70 +40,294 @@ Her har vi utelatt for høy inntekt (SU<0) og su under minstegrense (SU<2%)
-     object UtenlandsoppholdOver90Dager : Inngangsvilkår()
-     object InnlagtPåInstitusjon : Inngangsvilkår()
-     */
-
-    fun tilOpphørsgrunn() = when (this) {
-        Uførhet -> Opphørsgrunn.UFØRHET
-        Formue -> Opphørsgrunn.FORMUE
-    }
+    object Flyktning : Inngangsvilkår()
+    object LovligOpphold : Inngangsvilkår()
+    object Institusjonsopphold : Inngangsvilkår()
+    object OppholdIUtlandet : Inngangsvilkår()
+    object PersonligOppmøte : Inngangsvilkår()
+    object FastOppholdINorge : Inngangsvilkår()
 }
 
-data class Vilkårsvurderinger(
-    val uføre: Vilkår.Uførhet,
-    val formue: Vilkår.Formue = Vilkår.Formue.IkkeVurdert,
-) {
+fun Set<Vilkår>.erLik(other: Set<Vilkår>): Boolean {
+    return count() == other.count() &&
+        mapIndexed() { index, vilkår -> other.toList()[index].erLik(vilkår) }
+            .all { it }
+}
 
-    // TODO jah: Valider at de vurderte uføre-periodene er de samme som de vurderte formue-periodene
-    private val vilkår = setOf(uføre, formue)
+fun Nel<Vurderingsperiode>.erLik(other: Nel<Vurderingsperiode>): Boolean {
+    return count() == other.count() &&
+        mapIndexed() { index, vilkår -> other[index].erLik(vilkår) }
+            .all { it }
+}
 
-    fun oppdaterStønadsperiode(stønadsperiode: Stønadsperiode): Vilkårsvurderinger =
-        this.copy(
-            uføre = this.uføre.oppdaterStønadsperiode(stønadsperiode),
-            formue = this.formue.oppdaterStønadsperiode(stønadsperiode),
-        )
+sealed class Vilkårsvurderinger {
+    abstract val vilkår: Set<Vilkår>
 
-    companion object {
-        val IkkeVurdert = Vilkårsvurderinger(
-            uføre = Vilkår.Uførhet.IkkeVurdert,
-            formue = Vilkår.Formue.IkkeVurdert,
-        )
+    val periode: Periode? by lazy {
+        vilkår.flatMap { vilkår ->
+            when (vilkår) {
+                is FastOppholdINorgeVilkår.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is FlyktningVilkår.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is Vilkår.Formue.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is InstitusjonsoppholdVilkår.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is LovligOppholdVilkår.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is OppholdIUtlandetVilkår.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is PersonligOppmøteVilkår.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                is Vilkår.Uførhet.Vurdert -> {
+                    vilkår.vurderingsperioder.map { it.periode }
+                }
+                else -> emptyList()
```

Comment:
Er det mye som ligger i elsen her?

Reply:
Det er alle varianter hvor vi ikke har gjort noen vurderinger (disse har ingen periode). En sjekk i `GrunnlagsdataOgVilkårsvurderinger` bruker denne perioden for å sjekke at data samsvarer.